react/README.md

```diff
@@ -362,6 +362,149 @@ export default class Candidate extends React.Component {
 
 **[⬆ back to top](#table-of-contents)**
 
+### `className` prop
+
+Avoid defining a `className` prop for a component as its intent can be ambiguous. Will it overwrite the `className` set on the component's top-level node? Will it be merged in with the `className` set on the top-level node?
+
+Ideally a parent component should not be controlling the styling of a child component, so specifying a child's `className` prop should be unnecessary. The child component should be 100% in charge of its visual display.
+
+However, you may need to control the layout/positioning of a child component within a parent component. In this situation, the best solution is to wrap the child component in a `<div>` or `<span>` that the parent _does_ control so that the positioning styling can be added to that wrapper node.
+
+In the limited cases where the wrapper node solution doesn't work, the child component can expose a `__containerClassName` prop, which the parent can specify to add layout-based CSS class(es) to. The prop begins with dunder (`__`) to indicate that it is an exceptional case so that in can easily be spotted in code review.
+
+```js
+// good (parent uses wrapper `<div>` to position child)
+
+// Child.js
+export default class Child extends React.Component {
+    // code for Child component
+}
+
+// Parent.js
+export default class Parent extends React.Component {
+    render() {
+        return (
+            <div className="parent">
+                <h2 className="parent__heading">Parent</h2>
+
+                <div className="parent__child">
+                    <Child />
+                </div>
+            </div>
+        );
+    }
+}
+
+
+// less-than-ideal (parent uses `__containerClassName` prop)
```

Comment:
is this totally unacceptable or acceptable sometimes? just want to be sure it's clear!